app/main.py

```diff
@@ -1,3 +1,55 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, (int, float)):
+            self.km += other
+        if isinstance(other, Distance):
+            self.km += other.km
+        return self
+
+    def __mul__(self, other: int | float) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: int | float) -> Distance:
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, km: float | Distance) -> bool:
+        if isinstance(km, Distance):
+            return self.km < km.km
+        return self.km < km
+
+    def __gt__(self, km: float | Distance) -> bool:
+        if isinstance(km, Distance):
+            return self.km > km.km
+        return self.km > km
+
+    def __eq__(self, km: float) -> bool:
+        if isinstance(km, Distance):
+            return self.km == km.km
+        return self.km == km
+
+    def __le__(self, km: float) -> bool:
+        if isinstance(km, Distance):
+            return self.km <= km.km
+        return self.km <= km
+
+    def __ge__(self, km: float) -> bool:
+        if isinstance(km, Distance):
+            return self.km >= km.km
+        return self.km >= km
```

Comment:
Change parameter name `km` to `other`. And fix annotations, because other not only float.

Reply:
Done